bitnami/metallb/templates/daemonset.yaml

```diff
@@ -0,0 +1,107 @@
+      {{- if .Values.speaker.podAnnotations }}
+      annotations: {{- toYaml .Values.speaker.podAnnotations | nindent 8 }}
+      {{- end }}
+    spec:
+{{- include "metallb.speaker.imagePullSecrets" . | indent 6 }}
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        effect: NoSchedule
+      serviceAccountName: {{ include "metallb.speakerServiceAccountName" . }}
+      terminationGracePeriodSeconds: 2
+      hostNetwork: true
+      containers:
+      - name: speaker
+        image: {{ include "metallb.speaker.image" . }}
+        imagePullPolicy: {{ .Values.speaker.image.pullPolicy }}
+        args:
+        - --port={{ .Values.speaker.daemonset.hostPorts.metrics }}
+        - --config={{ include "metallb.configMapName" . }}
+        env:
+        - name: METALLB_NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        - name: METALLB_HOST
+          valueFrom:
+            fieldRef:
+              fieldPath: status.hostIP
+        - name: METALLB_ML_BIND_ADDR
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+        - name: METALLB_ML_LABELS
+          value: "app=metallb,component=speaker"
+        - name: METALLB_ML_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: METALLB_ML_SECRET_KEY
+          valueFrom:
+            secretKeyRef:
+              name: {{ include "metallb.fullname" . }}-memberlist
+              key: secretkey
+        ports:
+        - name: metrics
+          containerPort: {{ .Values.speaker.daemonset.hostPorts.metrics }}
+      {{- if .Values.speaker.livenessProbe.enabled }}
+        livenessProbe:
+          httpGet:
+            path: /metrics
+            port: metrics
+          initialDelaySeconds: {{ .Values.speaker.livenessProbe.initialDelaySeconds }}
+          periodSeconds: {{ .Values.speaker.livenessProbe.periodSeconds }}
+          timeoutSeconds: {{ .Values.speaker.livenessProbe.timeoutSeconds }}
+          successThreshold: {{ .Values.speaker.livenessProbe.successThreshold }}
+          failureThreshold: {{ .Values.speaker.livenessProbe.failureThreshold }}
+      {{- end }}
+      {{- if .Values.speaker.readinessProbe.enabled }}
+        readinessProbe:
+          httpGet:
+            path: /metrics
+            port: metrics
+          initialDelaySeconds: {{ .Values.speaker.readinessProbe.initialDelaySeconds }}
+          periodSeconds: {{ .Values.speaker.readinessProbe.periodSeconds }}
+          timeoutSeconds: {{ .Values.speaker.readinessProbe.timeoutSeconds }}
+          successThreshold: {{ .Values.speaker.readinessProbe.successThreshold }}
+          failureThreshold: {{ .Values.speaker.readinessProbe.failureThreshold }}
+      {{- end }}
+      {{- if .Values.speaker.resources }}
+        resources: {{- toYaml .Values.speaker.resources | nindent 10 }}
+      {{- end }}
+        securityContext:
+          runAsUser: 0
```

Comment:
The MetalLB speaker is a non-root container (https://github.com/bitnami/bitnami-docker-metallb-speaker/blob/124194ced06e3c1206d19ee5cf3fd7488f80a26c/0/debian-10/Dockerfile#L17), I still don't catch why it needs to `runAsRoot` here.

Regarding the securityContext you can see here the common way we use: https://github.com/bitnami/charts/blob/2ad44688f00899084a51d235c1f65b338d05429c/bitnami/redis-cluster/values.yaml#L140

Reply:
As I stated out in the referenced issue the speaker is not working when you deploy it as a nonRoot container. Because it needs access to the network interface on the host.

Reply:
https://github.com/bitnami/bitnami-docker-metallb-speaker/issues/1